File: Inxi.NET/Components/Processor.vb

'    Inxi.NET  Copyright (C) 2020-2021  EoflaOE
'
'    This file is part of Inxi.NET
'
'    Inxi.NET is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Inxi.NET is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Public Class Processor

    ''' <summary>
    ''' Name of processor
    ''' </summary>
    Public ReadOnly Name As String
    ''' <summary>
    ''' Core Topology
    ''' </summary>
    Public ReadOnly Topology As String
    ''' <summary>
    ''' Processor type
    ''' </summary>
    Public ReadOnly Type As String
    ''' <summary>
    ''' Processor milestone (Kaby Lake, Coffee Lake, ...)
    ''' </summary>
    Public ReadOnly Milestone As String
    ''' <summary>
    ''' Processor features
    ''' </summary>
    Public ReadOnly Flags() As String
    ''' <summary>
    ''' Processor bits
    ''' </summary>
    Public ReadOnly Bits As Integer
    ''' <summary>
    ''' L2 Cache
    ''' </summary>
    Public ReadOnly L2 As String
    ''' <summary>
    ''' L3 Cache
    ''' </summary>
    Public ReadOnly L3 As Integer
    ''' <summary>
    ''' CPU Rev
    ''' </summary>
    Public ReadOnly CPURev As String
    ''' <summary>
    ''' CPU BogoMips
    ''' </summary>
    Public ReadOnly CPUBogoMips As Integer
    ''' <summary>
    ''' Processor speed
    ''' </summary>
    Public ReadOnly Speed As String

    ''' <summary>
    ''' Installs specified values parsed by Inxi to the class
    ''' </summary>
    Friend Sub New(Name As String, Topology As String, Type As String, Bits As Integer, Milestone As String, Flags() As String, L2 As String, L3 As Integer, CPURev As String, CPUBogoMips As Integer, Speed As String)
        Me.Name = Name
        Me.Topology = Topology
        Me.Type = Type
        Me.Bits = Bits
        Me.Milestone = Milestone
        Me.Flags = Flags
        Me.L2 = L2
        Me.L3 = L3
        Me.CPURev = CPURev
        Me.CPUBogoMips = CPUBogoMips
        Me.Speed = Speed
    End Sub

End Class

File: Inxi.NET/Hardware/Processor.cs
﻿using InxiFrontend.Base;
using InxiFrontend.Core;
using Newtonsoft.Json;
using System;
using System.Linq;

namespace InxiFrontend
{
    /// <summary>
    /// Processor class
    /// </summary>
    public sealed class Processor : IHardware, IEquatable<Processor>
    {
        [JsonProperty()]
        /// <summary>
        /// Name of processor
        /// </summary>
        public string Name { get; set; }
        [JsonProperty()]
        /// <summary>
        /// Core Topology
        /// </summary>
        public string Topology { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// Processor type
        /// </summary>
        public string Type { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// Processor milestone (Kaby Lake, Coffee Lake, ...)
        /// </summary>
        public string Milestone { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// Processor features
        /// </summary>
        public string[] Flags { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// Processor bits
        /// </summary>
        public int Bits { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// L2 Cache
        /// </summary>
        public string L2 { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// L3 Cache
        /// </summary>
        public int L3 { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// CPU Rev
        /// </summary>
        public string CPURev { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// CPU BogoMips
        /// </summary>
        public int CPUBogoMips { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// Processor speed
        /// </summary>
        public string Speed { get; private set; }

        /// <summary>
        /// Installs specified values parsed by Inxi to the class
        /// </summary>
        internal Processor(string Name, string Topology, string Type, int Bits, string Milestone, string[] Flags, string L2, int L3, string CPURev, int CPUBogoMips, string Speed)
        {
            this.Name = Name;
            this.Topology = Topology;
            this.Type = Type;
            this.Bits = Bits;
            this.Milestone = Milestone;
            this.Flags = Flags;
            this.L2 = L2;
            this.L3 = L3;
            this.CPURev = CPURev;
            this.CPUBogoMips = CPUBogoMips;
            this.Speed = Speed;
        }
        [JsonConstructor()]
        public Processor()
        {

        }

        public bool Equals(Processor other)
        {
            return HelperFunctions.AreObjectsEqual<Processor>(this, other, (x) => x.CustomAttributes.Any(y => y.AttributeType == typeof(JsonPropertyAttribute)));
        }

        public override bool Equals(object obj)
        {
            if (obj is not Processor)
            {
                return false;
            }
            return this.Equals((Processor)obj);
        }

        public override int GetHashCode()
        {
            return HelperFunctions.GetHashCodes(this, (x) => x.CustomAttributes.Any(y => y.AttributeType == typeof(JsonPropertyAttribute)));
        }
    }
}